File: frontend/src/types/wallpaper.ts
export interface Wallpaper {
  id: string;
  name: string;
  path: string;
  thumbnail?: string;
  favorite: boolean;
  active: boolean;
  sizeBytes?: number;
  width?: number;
  height?: number;
  fileType?: string;
  createdAt?: string;
}

File: src-tauri/src/models/wallpaper.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Wallpaper {
    pub id: String,
    pub name: String,
    pub path: String,
    pub thumbnail: Option<String>,
    pub favorite: bool,
    pub active: bool,
    pub size_bytes: Option<u64>,
    pub created_at: Option<String>,
}